### ISSUE TITLE
Variadic macro expansion with comments

### ISSUE THREAD

**Issue by kanje** [doxygen/doxygen#10960]
This is somewhat an extension to https://github.com/doxygen/doxygen/issues/8542.

I want to use inline comments to document enum values defined with a macro such as
```cpp
/**
 * Enum with values.
 */
ENUM(Values,
    Value1, ///< First value.
    Value2, ///< Second value.
);
```

I did
```ini
PREDEFINED = "ENUM(t,...)=enum class t { __VA_ARGS__ }"
```
and this is what I am getting with `doxygen -d preprocessor`:
```
/**
 * Enum with values.
 */

///< First value.
///< Second value.
 enum class  Values  {  Value1,              Value2  };
```

I tried this with both Doxygen 1.9.5 and the latest master (1.12.0) --- the result is the same.

Would it be possible to somehow expand the macro while preserving the position of comments?

**Comment by albert-github** [comment]
An argument ends with the `,` so I thought maybe reorganizing the places of the `,` a little bit like:
```
/**
 * Second enum with values.
 */
ENUM(Values2
    , Value1_2 ///< First value.
    , Value2_2 ///< Second value.
);
```
might help but this gives the same results:
```
/**
 * Second enum with values.
 */

///< First value.
///< Second value.
 enum class  Values2  {  Value1_2, Value2_2  } ;
```

When using the `/**<` / `*/` type comments like:
```
/**
 * Third enum with values.
 */
ENUM(Values3
    , Value1_3 /**< First value. */
    , Value2_3 /**< Second value. */
);
```
we get
```
///< First value.
///< Second value.
 enum class  Values2  {  Value1_2, Value2_2  } ;
/**
 * Third enum with values.
 */



 enum class  Values3  {  Value1_3 /**< First value. */, Value2_3 /**< Second value. */  } ;
```
and the result is:
![image](https://github.com/doxygen/doxygen/assets/5223533/1ae14bfa-78b3-40f5-96cf-49a47ce6482f)

which looks OK to me (also the original formulation with `/**<` instead of `///<` looks OK).

All in all there looks like to be a difference in handling `/**<` and `///<` comments.

Example: 

As a side note  "the latest master (1.12.0)" is a moving target better would be to specify the complete output of `doxygen -v`.
(I tried with 1.12.0 (38ee114a4563fd91ea547c262d120cf361b83039))

**Comment by kanje** [comment]
Indeed, replacing `///<` with `/**<` and `*/` works around the issue, also for 1.9.5. Thanks!

> As a side note "the latest master (1.12.0)" is a moving target better would be to specify the complete output of doxygen -v.
(I tried with 1.12.0 (https://github.com/doxygen/doxygen/commit/38ee114a4563fd91ea547c262d120cf361b83039))

Point taken!

**Comment by doxygen** [comment]
@kanje Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by kanje** [comment]
@doxygen The problem is fixed, thanks!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
